### PR TITLE
chore: add ApplicationInsightsConfiguration, update removalPolicy

### PR DESCRIPTION
**Pull request by insignias**
Issue #, if available:

Description of changes:

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by DaveMadsen — APPROVED** [review]

File: workbench-core/infrastructure/src/workbenchAppRegistry.ts [hunk 1]
```diff
@@ -61,6 +65,10 @@ export class WorkbenchAppRegistry extends Construct {
       description: `Service Catalog application to track and manage all your resources for the solution ${this._solutionName}`
     });
 
+    if (this._destroy) {
```
Comment (htztomic) [inline workbench-core/infrastructure/src/workbenchAppRegistry.ts:68]: reason we apply removal policy twice?

Comment (insignias) [reply workbench-core/infrastructure/src/workbenchAppRegistry.ts:68]: @htztomic one is for application and the other is for attributeGroup

Comment (htztomic) [reply workbench-core/infrastructure/src/workbenchAppRegistry.ts:68]: got it!

File: workbench-core/infrastructure/src/workbenchAssociateStackToExistingAppRegistryApplication.ts [hunk 2]
```diff
@@ -31,5 +32,6 @@ export class WorkbenchAssociateStackToExistingAppRegistryApplication extends Con
     );
 
     importedApplication.associateApplicationWithStack(stack);
+    createAppInsightsConfiguration(stack);
```
Comment (FernandoAranda) [inline workbench-core/infrastructure/src/workbenchAssociateStackToExistingAppRegistryApplication.ts:35]: Is there any limitation of the amount of cfn applications we can create or associate to a stack ?

Comment (FernandoAranda) [inline workbench-core/infrastructure/src/workbenchAssociateStackToExistingAppRegistryApplication.ts:35]: is there any limitation in the number of CfnApplications we can create?

Comment (FernandoAranda) [reply workbench-core/infrastructure/src/workbenchAssociateStackToExistingAppRegistryApplication.ts:35]: any time we associate the App Registry to a stack we will be creating one

**Review by FernandoAranda — APPROVED** [review]

File: workbench-core/infrastructure/src/workbenchAssociateStackToExistingAppRegistryApplication.ts (hunk 2 above)
Comment (insignias) [reply workbench-core/infrastructure/src/workbenchAssociateStackToExistingAppRegistryApplication.ts:35]: Application Insights creates one CloudFormation stack for each application to manage CloudWatch agent installation and configuration for all application components. By default, each AWS account can have 2000 stacks.

**Review by htztomic — APPROVED** [review]

**Pull request merged**